### PR TITLE
Ionosphere tracing across fsgrid tasks

### PR DESCRIPTION
**Pull request by ursg**
This PR implements the loop to perform tracing in multiple stages, remapping the tracing from one MPI task to the next as they enter and leave fsgrid domains.

In principle, this can be used to trace using the perturbed B field on fsgrid (although the actual field access hasn't been implemented yet).

### CODE REVIEW

**Comment by ursg** [comment]
Current state of this pull request:
- There's a hole at the north pole (at node 0) when coupling a decently-sized simulation. This occurred in a restart of EGM.
- The FACs order of magnitude seems completely off. Yann is testing whether it is simply a missing division by DX in the balsara reconstructed curlB code.

**Comment by ykempf** [comment]
- The holes and blips are gone in Magnetosphere_3D_small and EGM restart courtesy of the removal of a nested `pragma omp parallel`. I had seen it many times and assumed this would be OK as @ursg had written it. :P
- The FAC order of magnitude seems better now that we divide by DX.

db10075 is bonus, but better safe than sorry?

File: fieldsolver/fs_common.cpp [hunk 1]
```diff
@@ -261,27 +261,38 @@ std::array<Real, 3> interpolatePerturbedB(
...
+
+   std::array<Real, Rec::N_REC_COEFFICIENTS> rc;
+   reconstructionCoefficients(
+      perBGrid,
+      dPerBGrid,
+      rc,
+      i,
+      j,
+      k,
+      3 // // Reconstruction order of the fields after Balsara 2009, 3 used to obtain 2nd order curl(B) and allows for cache reuse, see interpolatePerturbedB above
```
Comment (ykempf) [inline fieldsolver/fs_common.cpp:293]: Here actually now we can stick to 2.

**Comment by ursg** [comment]
Having run this at scale and in small tests, I'm now actually reasonable confident in the new tracing behaviour. Let's merge it into the ionosphere branch and continue there.

**Pull request merged**